framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -382,41 +397,44 @@ public void processClassTree(ClassTree classTree) {
             }
         }
 
+        AnnotatedTypeMirror classType = atypeFactory.getAnnotatedType(classTree);
+        QualifierHierarchy qualifierHierarchy = atypeFactory.getQualifierHierarchy();
         // If "@B class Y extends @A X {}", then enforce that @B must be a subtype of @A.
         // classTree.getExtendsClause() is null when there is no explicitly-written extends clause,
         // as in "class X {}". This is equivalent to writing "class X extends @Top Object {}", so
         // there is no need to do any subtype checking.
         if (classTree.getExtendsClause() != null) {
-            AnnotatedTypeMirror extendsClauseType =
-                    atypeFactory.getAnnotatedType(classTree.getExtendsClause());
-            checkExtendsOrImplementsClause(
-                    classTree, extendsClauseType, "declaration.inconsistent.with.extends.clause");
+            Set<AnnotationMirror> extendsAnnos =
```

Comment:
Is there a reason to separate this validation from the one above? This now iterates twice over the interfaces.
Maybe we need two methods, one for the basic validation done above and a second one for this `declaration.inconsistent.with.extends.clause` check.
See my comment on the GuiEffectVisitor why this might be necessary.

Reply:
I think these should just stay in processClassTree.  Otherwise it's too confusing on which method one should override.

Reply:
That's also not very nice, but let's do that for now.